File: POO/C#/Universidad.cs
using System;
using System.Text;

public class Universidad
{
    private string nombre;
    protected int estudiantes;
    public bool publica;
    private string infoJSON;
    private byte[] codigoAcreditacion;

    public Universidad(string nombre, int estudiantes, bool publica)
    {
        this.nombre = nombre;
        this.estudiantes = estudiantes;
        this.publica = publica;
        this.infoJSON = $"{{\"nombre\":\"{nombre}\",\"estudiantes\":{estudiantes}}}";
        this.codigoAcreditacion = Encoding.UTF8.GetBytes("COD-" + nombre);
    }

    public void AbrirPuertas()
    {
        Console.WriteLine("La universidad ha abierto sus puertas.");
    }

    public void MatricularEstudiantes(int nuevos)
    {
        estudiantes += nuevos;
    }

    public int ObtenerCantidad()
    {
        return estudiantes;
    }

    public string EvaluarRanking(int posicion)
    {
        return posicion <= 10 ? "Universidad destacada." : "Universidad promedio.";
    }

    public string Nombre
    {
        get { return nombre; }
        set { nombre = value; }
    }

    public string MostrarInfo()
    {
        return $"Universidad: {nombre}, Estudiantes: {estudiantes}, Pública: {publica}";
    }
}

File: POO/GO/Universidad.go
package main

import (
	"encoding/json"
	"fmt"
)

type Universidad struct {
	nombre            string
	estudiantes       int
	Publica           bool
	infoJSON          string
	codigoAcreditacion []byte
}

func NuevaUniversidad(nombre string, estudiantes int, publica bool) Universidad {
	return Universidad{
		nombre:             nombre,
		estudiantes:        estudiantes,
		Publica:            publica,
		infoJSON:           fmt.Sprintf(`{"nombre":"%s","estudiantes":%d}`, nombre, estudiantes),
		codigoAcreditacion: []byte("COD-" + nombre),
	}
}

func (u Universidad) AbrirPuertas() {
	fmt.Println("La universidad ha abierto sus puertas.")
}

func (u *Universidad) MatricularEstudiantes(nuevos int) {
	u.estudiantes += nuevos
}

func (u Universidad) ObtenerCantidad() int {
	return u.estudiantes
}

func (u Universidad) EvaluarRanking(posicion int) string {
	if posicion <= 10 {
		return "Universidad destacada."
	}
	return "Universidad promedio."
}

func (u *Universidad) GetNombre() string {
	return u.nombre
}

func (u *Universidad) SetNombre(n string) {
	u.nombre = n
}

func (u Universidad) MostrarInfo() string {
	return fmt.Sprintf("Universidad: %s, Estudiantes: %d, Pública: %t", u.nombre, u.estudiantes, u.Publica)
}
